Captured proposed implementation of ISPO working group content (#54)
InnerSourceCommons/ispo-working-group: README.md

These implementations will likely be documents, videos, or code, each of which is generic enough to be used across companies,
yet specific enough that people in the company can leverage them to solve their problems without needing notable amounts of company-specific work.

## Implementation

> We expect to create together The Handbook of InnerSource, which contains our collective, practical knowledge of how to run InnerSource projects and programs.

Reviewer: ```suggestion
We expect to create together **The Handbook of InnerSource**, which contains our collective, practical knowledge of how to run InnerSource projects and programs.
```

The goal is that a new practioner can implement the instruction in the handbook and be successful with it in their InnerSource goals.
We expect that the handbook will have various sections for the different personas and roles involved with InnerSource, e.g. ISPO group, engineering managers, product leaders, individual engineers, etc.
The handbook may look something like the [Managing InnerSource Projects](https://innersourcecommons.gitbook.io/managing-inner-source-projects/) book and, in fact, may end up being a superset and subsume the content there.

We imagine that content in the handbook will consist of 3 parts:
1. A general pattern describing, at a high level, what to do for some aspect of InnerSource.
This is important so that the general principles and approaches of InnerSource are captured.

> This part of the content may be similar to the conten tin the existing [InnerSource Patterns](https://innersourcecommons.org/learn/books/innersource-patterns/) book.

Reviewer: ```suggestion
This part of the content may be similar to the content in the existing [InnerSource Patterns](https://innersourcecommons.org/learn/books/innersource-patterns/) book.
```

1. (Multiple) examples of how that pattern is implemented in different contexts.
These concrete examples are important because they allow practioners to select the implementation that matches their context and then execute the described implementation.
We don't want practitioners to have the overhead of figuring out over and over again how to adapt a pattern to their given context.
1. Explanatory videos that describe the general pattern or its example implementations (separate vidoes for each).
Ideally every text article in the handbook has an accompanying video.
People like to watch training videos, so it makes the content more accessible to them.

> Once there is a concrete example of some content that contains these 3 parts, then we will link to it here as an example (tracked via [#53](https://github.com/InnerSourceCommons/ispo-working-group/issues/53).

Reviewer: ```suggestion
Once there is a concrete example of some content that contains these 3 parts, we will link to it here as an example (tracked via [#53](https://github.com/InnerSourceCommons/ispo-working-group/issues/53)).
```

These implementations will likely be documents, videos, or code, each of which is generic enough to be used across companies,
yet specific enough that people in the company can leverage them to solve their problems without needing notable amounts of company-specific work.

## Implementation

We expect to create together **The Handbook of InnerSource**, which contains our collective, practical knowledge of how to run InnerSource projects and programs.
The goal is that a new practioner can implement the instruction in the handbook and be successful with it in their InnerSource goals.
We expect that the handbook will have various sections for the different personas and roles involved with InnerSource, e.g. ISPO group, engineering managers, product leaders, individual engineers, etc.

> The handbook may look something like the [Managing InnerSource Projects](https://innersourcecommons.gitbook.io/managing-inner-source-projects/) book and, in fact, may end up being a superset and subsume the content there.

Reviewer: Maybe move this sentence to the "some comments on this approach" section.
It is a helpful side node for people that already know the other ISC resources but does not seem essential to understand the general idea of the approach suggested here.

Author: Good idea.

We imagine that content in the handbook will consist of 3 parts:

> 1. A general pattern describing, at a high level, what to do for some aspect of InnerSource.
> This is important so that the general principles and approaches of InnerSource are captured.
> In the ideal state this part of the content would be shared with the existing [InnerSource Patterns](https://innersourcecommons.org/learn/books/innersource-patterns/) book.
> 1. (Multiple) examples of how that pattern is implemented in different contexts.
> These concrete examples are important because they allow practioners to select the implementation that matches their context and then execute the described implementation.
> We don't want practitioners to have the overhead of figuring out over and over again how to adapt a pattern to their given context.
> 1. Explanatory videos that describe the general pattern or its example implementations (separate vidoes for each).

Reviewer: ```suggestion
1. **A general pattern** describing, at a high level, what to do for some aspect of InnerSource.
This is important so that the general principles and approaches of InnerSource are captured.
In the ideal state this part of the content would be shared with the existing [InnerSource Patterns](https://innersourcecommons.org/learn/books/innersource-patterns/) book.
1. **(Multiple) examples** of how that pattern is implemented in different contexts.
These concrete examples are important because they allow practioners to select the implementation that matches their context and then execute the described implementation.
We don't want practitioners to have the overhead of figuring out over and over again how to adapt a pattern to their given context.
1. **Explanatory videos** that describe the general pattern or its example implementations (separate vidoes for each).
```

Formatting suggestion to highlight what the different parts are.

Author: Great formatting.  Thank you!

Reviewer 2: > 1. **A general pattern** describing, at a high level, what to do for some aspect of InnerSource.
> This is important so that the general principles and approaches of InnerSource are captured.
> In the ideal state this part of the content would be shared with the existing [InnerSource Patterns](https://innersourcecommons.org/learn/books/innersource-patterns/) book.
> 1. **(Multiple) examples** of how that pattern is implemented in different contexts.

Hi, reading this part, I can't understand what will differentiate the work done in ISPO WG from the work done for Inner Source Patterns. Maybe it's because I have never contributed to an "Inner Source Pattern," and I apologize for that. It seems that the patterns already have a "known instances" section. In the end, perhaps the content of ISPO could simply be new patterns that add rich content: concrete examples and videos ?

Ideally every text article in the handbook has an accompanying video.
People like to watch training videos, so it makes the content more accessible to them.

> 

Reviewer: You also mention "code" as a possible part of the implementation further up.
Should we add a 4th point here that captures anything related to code?

Author: I think that coding projects will be separate from the handbook project.  Added a small line to indicate that the handbook only is meant to address the teaching and training materials that we produce together.

Click through for a diagram showing what this might look like.

[<img src="https://github-production-user-asset-6210df.s3.amazonaws.com/9609562/245475861-de4e1b79-1d1c-41ad-8f2d-c1d612fed649.png" width="50%" />](https://miro.com/app/board/uXjVM_OGfjY=/)

Some comments on this approach:
* For a given article, not all 3 parts of it (pattern, implementation, video) need to be done at once.
The handbook should be able to render any combination of one, two, or three of these so that people can contribute whatever part(s) they're interested in.

> * If this approach works well and is useful, then it may subsume most of the content from the existing Learning Path and Pattenrs working groups.

Reviewer: ```suggestion
* If this approach works well and is useful, then it may subsume most of the content from the existing Learning Path and Patterns working groups.
```

> These implementations will likely be documents, videos, or code, each of which is generic enough to be used across companies,

Reviewer 3: I added a comment earlier, but noticed your request for me to review it.

Given our recent work, I'd like to conduct a review in a working session.

Author: Sounds good.
